client/src/helpers/qnAHelper.js

```diff
@@ -1,337 +1,94 @@
-      answers.forEach((ans, i) => {
-        if (index === i) {
-          question.answers = ans
-        }
-      })
-    })
+  createDynamicData(questions) {
 
-    let sortedQuestions = this.sortQuestions(copy)
-    let answersArr =[]
-    sortedQuestions.forEach((question) => {
-      answersArr.push(question.answers)
+    let finalQuestions = questions.sort((a, b) => {
+      return b.question_helpfulness - a.question_helpfulness
+      // return b = a
     })
-
-    let sortedAnswers = this.sortAnswers(answersArr)
-    // console.log([sortedQuestions, sortedAnswers])
+    let newQuestionsWithAnsArr = finalQuestions.filter((question) => {
+      let newAns = Object.values(question.answers)
+      let finalAns = this.sortAnswers(newAns)
+      question.answers = finalAns
+      return finalQuestions
+    })
+    return newQuestionsWithAnsArr
 
 
-    return [sortedQuestions, sortedAnswers]
   },
 
-  filterSearchInput(currentQuestions, searchTerm) {
-
 
+  filterSearchInput(currentQuestions, searchTerm) {
     let newQuestions = currentQuestions.filter((question) => {
       let lowerCaseQuestion = question.question_body.toLowerCase().replace(/\s+/g, '')
       let lowerCaseInput = searchTerm.toLowerCase().replace(/\s+/g, '')
 
       if (lowerCaseQuestion.includes(lowerCaseInput)) {
         return question
-
       }
-
     })
 
     return newQuestions
   },
-//need to fix more answered questions display
-  showMoreAnsweredQuestions(arr) {
-
-    let showButton;
-      if (arr[0].length > 2) {
-        showButton = true;
-      } else {
-        showButton = false
-      }
-      return showButton
-  },
 
   moreAnsweredQButtonDisplay(qClickCount, index) {
-    console.log(index)
-    console.log(qClickCount)
+    // console.log(index)
+    // console.log(qClickCount)
     let result = false;
     if (qClickCount === index || qClickCount + 1 === index) {
       result= true;
     }
     return result
   },
 
-  loadAnswerButtonText(currentCount) {
-    let text;
-
-    if (currentCount % 2 !== 0) {
-      text = 'Collapse Answers'
-
-    } else {
-      text = 'Load More Answers'
-    }
-    return text
-  },
-
-  qListScrollClass (count) {
-    let show;
-    if (count > 1) {
-      show = true
-    }
-    return show
-
-  },
-
-  answerHideClass(classname, index) {
-    let newClass;
-
-
-    if (index <= 1) {
-      newClass = 'answerListTable'
-    }
-    if (classname === 'Hide') {
-      newClass = 'answerListTable Hide'
-    }
-
-    if (index >= 2) {
-      newClass = 'answerListTable Hide'
-    }
-    return newClass
-  },
-
-  answerTableHideClass(count, i) {
-    let newClass;
-    if (count % 2 !== 0)  {
-      if (i >= 2) {
-        newClass = 'answerListTable'
-
-      }
-
-    } else {
-
-      if (i >= 2) {
-        newClass = 'answerListTable Hide'
-
-      }
-    }
-    return newClass
-
-  },
-
   showQuestionsClass(clickCount, index) {
     let show= false;
+
     if (index <= clickCount) {
```

Comment:
Could condense with ternary if you wanted here. `index <= clickCount ? true : false`. Not sure if you need show to default to false though, that could potentially complicate